tests/unit_tests/test_core_api.py

```diff
@@ -103,6 +107,13 @@ def test_data_endpoint_returns_data(test_app):
     ]
 
 
+@mock.patch("importlib.metadata.version", new=fake_version)
+def test_version_endpoint_returns_version(test_app):
```

Comment:
When you're mocking functions it's a good practice, to check whether you called it.

Also - instead of creating function for that you can just use `return_value` :

```
@mock.patch("importlib.metadata.version", return_value="0.1.2")
def test_version_endpoint_returns_version(mock_returning_version, test_app):
```

and then inside - check if such mock was called

Reply:
Yes, I tried these things. But `return_value` does not work. I think if you are replacing a function, you need to specify him a function in replace.
And I don't know how to apply `assert_called_once` since it is a method of that mock object. I don't know what could be this mock object here (not `fake_version`, tested).
I didn't dive into mock so I don't know how these objects work.

Reply:
hmm... this code I just pasted - I tried this and it worked :)
I usually check my code first (not allways I have to admit, but usually).

Didn't want to give you whole solution on a plate, but...

```suggestion
@mock.patch("importlib.metadata.version", return_value="0.1.2")
def test_version_endpoint_returns_version(mock_returning_version, test_app):
    response = test_app.get("/api/version")
    mock_returning_version.assert_called_once_with("goodmap")
    assert response.status_code == 200
    assert response.json == {"backend": "0.1.2"}
```

Reply:
Now I understand what you said that metaprogamming is not an easy way